conjure/controllers/deploystatus/gui.py

```diff
@@ -23,18 +25,30 @@ def __fatal(error):
     return __handle_exception('ED', Exception(error))
 
 
-def finish():
+def __handle_exception(tag, exc):
+    utils.pollinate(app.session_id, tag)
+    app.ui.show_exception_message(exc)
+
+
+def __wait_for_applications_exception(exc):
+    return __handle_exception('ED', exc)
+
+
+def __wait_for_applications(*args):
     deploy_done_sh = os.path.join(this.bundle_scripts,
                                   '00_deploy-done.sh')
-    common.wait_for_applications(deploy_done_sh,
-                                 __fatal,
-                                 app.ui.set_footer)
-    return controllers.use('steps').render()
 
+    future = async.submit(partial(common.wait_for_applications,
+                                  deploy_done_sh,
+                                  __fatal,
+                                  app.ui.set_footer),
+                          __wait_for_applications_exception)
+    future.add_done_callback(finish)
 
-def __handle_exception(tag, exc):
-    utils.pollinate(app.session_id, tag)
-    app.ui.show_exception_message(exc)
+
+def finish(future):
+    future.result()
```

Comment:
why call result() here? wait_for_applications doesn't actually return anything anyway, unless there's an error (and in that case it just returns the result of calling the callback, which isn't going to be anything useful anyway)